roscala/src/main/scala/coop/rchain/rosette/Ctxt.scala

```diff
@@ -1,42 +1,91 @@
 package coop.rchain.rosette
 
+import coop.rchain.rosette.Location.StoreCtxt
+import Location._
 import scala.collection.mutable
 
-case class Ctxt(argvec: Tuple,
-                code: Code,
-                ctxt: Ctxt,
-                env: Ob,
-                monitor: Monitor,
+case class Ctxt(tag: Location,
                 nargs: Int,
                 outstanding: Int,
                 pc: PC,
-                reg: Seq[Ob],
-                rslt: Ob,
-                trgt: Ob,
-                selfEnv: Ob,
-                override val _slot: mutable.Seq[Ob],
-                tag: Location)
+                rslt: Ob, // reg[0]
+                trgt: Ob, // reg[1]
+                argvec: Tuple, // reg[2]
+                env: Ob, // reg[3]
+                code: Code, // reg[4]
+                ctxt: Ctxt, // reg[5]
+                self2: Ob, // reg[6]
+                selfEnv: Ob, // reg[7]
+                rcvr: Ob, // reg[8]
+                monitor: Monitor, // reg[9]
+                override val _slot: mutable.Seq[Ob])
     extends Ob {
+  def applyK(result: Ob, tag: Location)(state: VMState): (Boolean, VMState) =
+    ctxt.rcv(result, tag)(state)
+
   def arg(n: Int): Option[Ob] = argvec.elem.lift(n)
 
+  def getReg(r: Int): Option[Ob] =
+    Vector(rslt, trgt, argvec, env, code, ctxt, self2, selfEnv, rcvr, monitor)
```

Comment:
Does this create a new vector each time, or is the compiler smart enough to cache it?

Reply:
Seems like the compiler is not smart enough.